### PR TITLE
Share task text with carrot template engine. #372

### PR DESCRIPTION
**Pull request by lemonboston**
This pull request replaces the 'manual' assembly of the shared task text with using the carrot template engine.

---

@dmfs I am submitting this now but there are TODOs left listed in `ShareTaskText`, mostly related to `androidcarrot` about the issues opened [there](https://github.com/dmfs/androidcarrot/issues). Shall I create pull requests for those?

### CODE REVIEW

**Comment by dmfs** [comment]
let's wait for the next carrot release. All our changes have been merged so we don't need to use our own fork.

**Comment by lemonboston** [comment]
We don't use a fork any more here, `carrot` is imported directly with commit hash with jitpack.
But we use `AndroidBindings` from your `androidcarrot` extension library. Shall we copy the necessary files from that to OpenTasks?

The related pending TODOs are listed here:
https://github.com/dmfs/opentasks/blob/174c1871651f6578590cf4f0b3f5e8590cd2ee8e/opentasks/src/main/java/org/dmfs/tasks/share/ShareTaskText.java#L67-L67

File: opentasks/proguard.cfg [hunk 1]
```diff
@@ -77,4 +77,6 @@
     java.lang.String TAG;
     @org.dmfs.android.retentionmagic.annotations.* <fields>;
     private long mId;
-}
\ No newline at end of file
+}
+
+-dontwarn au.com.codeka.carrot.**
```
Comment (dmfs) [inline opentasks/proguard.cfg]: is this still necessary?

Comment (lemonboston) [reply opentasks/proguard.cfg]: I was just checking that, no there were no more warnings, so I've removed it.

**Comment by lemonboston** [comment]
Ready for review now.
There are those `Lazy`, `Factory` types added here which may be replaced with the final ones I assume coming at some point in `jems`.

**Pull request merged**